cli/azd/extensions/microsoft.azd.extensions/internal/github/github.go

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package github
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/extensions/microsoft.azd.extensions/internal"
+	"github.com/azure/azure-dev/cli/azd/pkg/common"
+)
+
+var (
+	ErrReleaseAlreadyExists = fmt.Errorf("release already exists")
+	ErrReleaseNotFound      = fmt.Errorf("release not found")
+)
+
+// GitHubCli provides access to GitHub CLI functionality
+type GitHubCli struct {
```

Comment:
Why are we creating a copy of this tool instead of leveraging how AZD can handle this gh cli?  
AZD knows how to auto-install, login, and more operations

Reply:
See comments above.